corehq/apps/app_manager/dbaccessors.py

```diff
@@ -288,21 +290,24 @@ def get_latest_app_ids_and_versions(domain, app_id=None):
 
 def get_all_apps(domain):
     """
-    Returns a list of all the apps ever built and current Applications.
+    Returns an iterable over all the apps ever built and current Applications.
     Used for subscription management when apps use subscription only features
     that shouldn't be present in built apps as well as app definitions.
     """
-    from .models import Application
-    from corehq.apps.app_manager.util import get_correct_app_class
-    saved_apps = Application.get_db().view(
-        'app_manager/saved_app',
-        startkey=[domain],
-        endkey=[domain, {}],
-        include_docs=True,
-    )
-    all_apps = [get_correct_app_class(row['doc']).wrap(row['doc']) for row in saved_apps]
-    all_apps.extend(get_apps_in_domain(domain))
-    return all_apps
+    def _saved_apps():
+        from .models import Application
+        from corehq.apps.app_manager.util import get_correct_app_class
+        saved_app_ids = Application.get_db().view(
+            'app_manager/saved_app',
+            startkey=[domain],
+            endkey=[domain, {}],
+            include_docs=False,
+            wrapper=lambda row: row['id'],
+        )
+        correct_wrap = lambda app_doc: get_correct_app_class(app_doc).wrap(app_doc)
+        return imap(correct_wrap, iter_docs(Application.get_db(), saved_app_ids))
+
+    return chain(get_apps_in_domain(domain), _saved_apps())
```

Comment:
Looks from the commit history that you ran this and found an error? Just confirming (since there still aren't software tests) that you tested this function manually.